src/machine.rs

```diff
@@ -253,6 +253,23 @@ impl MemoryExtra {
                 this.write_scalar(Scalar::from_u8(0), &place.into())?;
                 Self::add_extern_static(this, "_tls_used", place.ptr);
             }
+            "android" =>
+                for symbol_name in &["signal", "bsd_signal"] {
+                    let layout = this.machine.layouts.usize;
+                    let dlsym = Dlsym::from_str(symbol_name.as_bytes(), &this.tcx.sess.target.os)?
+                        .unwrap_or_else(|| {
+                            panic!(
+                                "hardcoded `extern static` symbol {} has no dlsym handler",
+                                symbol_name
+                            )
+                        });
+                    let (provenance, offset) =
+                        this.memory.create_fn_alloc(FnVal::Other(dlsym)).into_parts();
```

Comment:
Taking the pointer apart here doesn't seem right. `write_pointer` should accept what `create_fn_alloc` returns.

Reply:
`write_pointer` I *think* does, but `add_extern_static` does not. It didn't feel right to me either.

Reply:
`ptr.into()` should work for `add_extern_static`.

If there are other things that didn't feel right, please let me know. :)